backends/tfhe-cuda-backend/cuda/src/integer/scalar_comparison.cuh

```diff
@@ -843,13 +839,13 @@ __host__ void host_integer_radix_scalar_equality_check_kb(
     are_all_comparisons_block_true<Torus>(
         streams, gpu_indexes, gpu_count, lwe_array_out,
         mem_ptr->tmp_lwe_array_out, mem_ptr, bsks, ksks,
-        num_halved_scalar_blocks + (num_msb_radix_blocks > 0));
+        num_halved_scalar_blocks + msb_out.num_radix_blocks);
```

Comment:
Now we need to check for all blocks in the output of host_compare_with_zero, instead of just 1 block.